### DISCUSSION TITLE
Welcome to abqpy Discussions!

### DISCUSSION THREAD

**Discussion by haiiliin** [haiiliin/abqpy#146] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by abqlearner** [comment]
Hello, I tried to run my abaqus script with loop submission. I met this problem when the sceond job sumitted:
WARNING: Interactive messaging has been requested for this analysis run, 
             but initialization of the messaging system has failed.  Therefore 
             no messages will be sent to Abaqus/CAE for this phase of the 
             analysis.
How can I fix this question？
![a48f474101e70718483180f739b3207](https://github.com/haiiliin/abqpy/assets/163389919/75c9596a-4dbd-404e-a83d-d5b274503fa6)

**Reply by haiiliin** [reply]
Try to call `job.waitForCompletion()` method after each job is submitted:

```python
job = mdb.jobs['Job-reli' + str(mod_num)]
job.submit(consistencyChecking=OFF)
job.waitForCompletion()
```

**Reply by abqlearner** [reply]
It helps. Thanks.

**Comment by thelongip** [comment]
Hello everyone, I'm new to Github (first time). 
I'm writting this beacause i think i encontered a bug so i'd like to report it. 
I explain. 

I installed abqpy-2020.7.1 (with abaqus2020 and python 3.10.11). 

when i want to launch my code (code below) i have the following error :
"""
The script will be submitted to Abaqus next and the current Python session will be closed.
Traceback (most recent call last):
  File "worspace_path\test.py", line 1, in <module>
    from abaqus import mdb
  File "C:\Users\user_name\AppData\Local\Programs\Python\Python310\lib\site-packages\abaqus\__init__.py", line 7, in <module>
    run(cae=True)
  File "C:\Users\user_name\AppData\Local\Programs\Python\Python310\lib\site-packages\abqpy\run.py", line 48, in run
    abaqus.cae(filePath, *sys.argv[1:], **config.cae.model_dump())
TypeError: AbqpyCLI.cae() got an unexpected keyword argument 'startup'
"""

I also tried with another version of abqpy-2020 (6.2) and it worked perfectly.

The cae function (from abqpy>cli.py>class AbqpyCLI) has not 'startup' in its keywords (in both versions). So the problem could come from one of the inputs of "abaqus.cae(filePath, *sys.argv[1:], **config.cae.model_dump())" but my knowledge stops here. 

I hope it helps. 
Have a great day, 
Théo




all is executed on an offline windows 10

## Python code : test.py ##
from abaqus import mdb

job = mdb.JobFromInputFile('job_name', 'job_name.inp')
job.submit()
job.waitForCompletion()

## More Things related ##

py file executed in vscode
results from pip list : 
abqpy             2020.7.1
auto-all          1.4.1
fire              0.6.0
pip               23.0.1
pydantic          1.10.14
setuptools        65.6.3
six               1.16.0
termcolor         2.4.0
typeguard         4.1.5
typing_extensions 4.10.0

**Reply by haiiliin** [reply]
Thanks for reporting, it is a bug that only occurred in version 7.1

**Reply by haiiliin** [reply]
New version 7.2 has been released that fixes this issue

**Comment by abqlearner** [comment]
There are two questions for me.

Q1: I uninstall abaqus2022 and install abaqus2024. However, the command was run in abaqus.bat of abaqus2022. How can I change its address?
 
Q2: I want to use some algorithms in scikit_learn to optimize some parameters of the model (based on simulation results). How can I correctly install sklearn and use relevant functions in the script?
It seems that "import sklearn" cannot be run in cae, even though I have already installed this library for python in abaqus2024.

**Reply by haiiliin** [reply]
Q1: https://haiiliin.github.io/abqpy/en/2024/envvars.html#envvar-ABAQUS_BAT_PATH
Q2: https://haiiliin.github.io/abqpy/en/2024/examples/Parameter-Identification/optimize.html#sphx-glr-examples-parameter-identification-optimize-py

**Reply by Crook-cyber** [reply]
same q2
